### PR TITLE
feat(api): add gNMI network state endpoint

### PR DESCRIPTION
**Pull request by vihu**
# Summary

Adds a read-only `/api/dz/network-state` endpoint that summarizes existing gNMI telemetry for the selected DoubleZero environment.

## Changes

- Add env-aware network state API handler backed by telemetry ClickHouse tables.
- Summarize telemetry freshness across interface, BGP, ISIS, and transceiver tables.
- Aggregate latest interface state by interface family, including `Switch*` fabric interfaces.
- Aggregate latest BGP neighbors and ISIS adjacencies by raw OpenConfig state.
- Add optics volume summary from latest transceiver state and threshold telemetry.
- Include known-gap flags for empty telemetry, mainnet pilot gaps, and empty ISIS overload-bit data.
- Register `GET /api/dz/network-state` route.
- Add handler tests for empty telemetry, missing telemetry schema, and populated aggregate responses.

## Validation

- `go test ./api/handlers -run 'TestGetNetworkState|TestGetDeviceOptics' -count=1`
- `go test ./api/handlers -count=1`
- `make fmt`
- `make lint`

### CODE REVIEW

File: api/handlers/network_state.go [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		logError("network state: failed to encode response", "error", err)
+	}
+}
+
+func queryNetworkFreshness(ctx context.Context, conn driver.Conn, tdb string) ([]TelemetryFreshness, error) {
+	freshness := make([]TelemetryFreshness, 0, len(networkStateFreshnessTables))
+	for _, table := range networkStateFreshnessTables {
```
Comment (ben-dz) [inline api/handlers/network_state.go]: Could this loop make a single query or parallelize rather than so many sequential ones all inside the same context timeout within GetNetworkState?

Comment (vihu) [reply api/handlers/network_state.go]: Right, good call. Could get away with a `union all` query here I think. I'll make that change shortly.

Comment (vihu) [reply api/handlers/network_state.go]: Updated: [e23b8d6](https://github.com/malbeclabs/lake/pull/606/commits/e23b8d61c5055354b8613b8a33866bae55c81478)

**Review by ben-dz — APPROVED** [review]
lgtm
